draft-ietf-gnap-core-protocol.md

```diff
@@ -204,29 +194,37 @@ role of the AS as defined by the protocol.
 In addition to the roles above, the protocol also involves several 
 elements that are acted upon by the roles throughout the process.
 
+Attribute
+: characteristics related to an end-user.
+
 Access Token
-: A credential representing a set of access rights
-delegated to the client instance. The access token is created by the AS, consumed
-and verified by the RS, and issued to and carried by the client instance. The contents
-and format of the access token are opaque to the client.
+: digitally signed data that contains specific rights and/or attributes.
```

Comment:
A token needn't be signed and needn't contain any information in it whatsoever. Reference-based tokens have neither, for example. Suggest we keep the wider definition. 

```suggestion
: a data artifact representing a set of delegated access rights, the format of which is opaque to the client software.
```

Reply:
I don't think we should include the "opaque" part here: this is an implementation detail and even if specified by the protocol, it is not part of the definition of "token". If we ever choose to support non-opaque tokens, there's no reason to change this definition.

Reply:
Opaqueness to the client is a clear line in terms of roles and responsibilities. The token value is a message from the AS to the RS, which the client carries but is not required to understand. If there is other information the client needs, it needs to be sent separately from the token. This is where the "subject information" and assertions and the like come into play. As soon as we say "the token might not be opaque" we need to define what the format and contents of the token are, and set expectations of what we expect clients to understand within the token in order to function. The fact that the client doesn't have to see inside the token is one of the key benefits of OAuth that we would be remiss to throw out here.

Note that the token is only opaque to the client, it is not opaque to the RS or AS.

Reply:
This is all good, I just think it is inappropriate in a *definition*. It's an important property of the token, yes, but it's not part of its definition IMHO.

Reply:
Yes, we tried to avoid having definitions that embed requirements (here opaque). We should put that in another section.

Reply:
proposal : a data artifact representing a set of rights and/or attributes.

again we have the "delegated" part which would need a definition if we want to include that. I'll think about it.

Reply:
Currently left a note in issue #145, resolving this conversation for now